src/main/java/net/javadiscord/javabot/systems/moderation/AutoMod.java

```diff
@@ -29,15 +34,29 @@
 @Slf4j
 public class AutoMod extends ListenerAdapter {
 
-	private static final Pattern inviteURL = Pattern.compile("discord(?:(\\.(?:me|io|gg)|sites\\.com)/.{0,4}|app\\.com.{1,4}(?:invite|oauth2).{0,5}/)\\w+");
+	final Pattern inviteURL = Pattern.compile("discord(?:(\\.(?:me|io|gg)|sites\\.com)/.{0,4}|app\\.com.{1,4}(?:invite|oauth2).{0,5}/)\\w+");
+	Pattern urlPattern = Pattern.compile(
+			"(?:^|[\\W])((ht|f)tp(s?)://|www\\.)"
+					+ "(([\\w\\-]+\\.)+?([\\w\\-.~]+/?)*"
+					+ "[\\p{Alnum}.,%_=?&#\\-+()\\[\\]*$~@!:/{};']*)",
+			Pattern.CASE_INSENSITIVE | Pattern.MULTILINE | Pattern.DOTALL);
 	private List<String> spamUrls;
 
 	public AutoMod() {
 		try {
-			spamUrls = Arrays.stream(StringResourceCache.load("/spamLinks.txt").split(System.lineSeparator())).toList();
-		} catch (Exception e) {
+			URL url = new URL("https://raw.githubusercontent.com/DevSpen/scam-links/master/src/links.txt");
+			HttpsURLConnection connection = (HttpsURLConnection) url.openConnection();
+			InputStream stream = connection.getInputStream();
+			String response = new Scanner(stream).useDelimiter("\\A").next();
+			spamUrls = List.of(response.split("\n"));
+		} catch (IOException e) {
 			e.printStackTrace();
-			spamUrls = List.of();
+			try {
+				spamUrls = Arrays.stream(StringResourceCache.load("/spamLinks.txt").split(System.lineSeparator())).toList();
+			} catch (Exception exception) {
+				exception.printStackTrace();
+				spamUrls = List.of();
+			}
```

Comment:
I think you can remove that and the `spamLinks.txt` file entirely.

Reply:
well, what if due to some exception we were not able to fetch from url, we are still safe to some extend so it might not be a good idea to remove that file